packages/core/src/UserState.ts

```diff
@@ -25,39 +25,59 @@ import { Synchronizer, toDecString } from './Synchronizer'
  * It takes user's `ZKIdentity` and checks the events that matches the user's identity.
  */
 export default class UserState {
+    prover: Prover
     public id: Identity
     public sync: Synchronizer
 
     get commitment() {
         return this.id.commitment
     }
 
-    constructor(
-        config:
-            | {
-                  db?: DB
-                  attesterId?: bigint | bigint[]
-                  unirepAddress: string
-                  prover: Prover
-                  provider: ethers.providers.Provider
-                  _id?: Identity // TODO: remove this and only accept as second arg
-              }
-            | Synchronizer,
+    constructor(config: {
+        synchronizer?: Synchronizer
+        db?: DB
+        attesterId?: bigint | bigint[]
+        unirepAddress?: string
+        provider?: ethers.providers.Provider
         id: Identity
-    ) {
-        if (config instanceof Synchronizer) {
-            if (!id) {
+        prover: Prover
+    }) {
+        const {
+            id,
+            synchronizer,
+            attesterId,
+            unirepAddress,
+            provider,
+            prover,
+            db,
+        } = config
+        if (!id) {
+            throw new Error(
+                '@unirep/core:UserState: id must be supplied as second argument when initialized with a sync'
+            )
+        }
+        if (!prover) {
+            throw new Error(
+                '@unirep/core:UserState: prover must be supplied as second argument when initialized with a sync'
+            )
+        }
+        if (synchronizer) {
+            this.sync = synchronizer
+        } else {
+            if (!provider || !unirepAddress) {
                 throw new Error(
-                    '@unirep/core:UserState: id must be supplied as second argument when initialized with a sync'
+                    '@unirep/core:UserState: provider and Unirep address must be supplied as second argument when initialized with a sync'
```

Comment:
This error message should not have the `supplied as second argument` text. There is only 1 argument now